### PR TITLE
replace confluent with samrama

### PR DESCRIPTION
**Pull request by lxning**

### CODE REVIEW

File: datanode/datanode_test.go [hunk 1]
```diff
@@ -44,38 +44,38 @@ var _ = ginkgo.Describe("datanode", func() {
...
-					Controller:  &common.ControllerConfig{Address: "localhost:6708"},
-					Etcd: etcd.Configuration{
-						Zone:    "local",
-						Env:     "test",
-						Service: "ares-datanode",
-						ETCDClusters: []etcd.ClusterConfig{
-							{
-								Zone:      "local",
-								Endpoints: []string{"127.0.0.1:2379"},
+				SetServerConfig(common.AresServerConfig{
```
Comment (jshencode) [inline datanode/datanode_test.go]: this indentation seems unnecessary

File: subscriber/common/consumer/kafka/kafka.go [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+	"github.com/uber/aresdb/subscriber/config"
+	"github.com/uber/aresdb/utils"
+	"go.uber.org/zap"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+var Test = false
```
Comment (voyager-dw) [inline subscriber/common/consumer/kafka/kafka.go]: this is a little weird

File: datanode/datanode_test.go (hunk 1 above)
Comment (lxning) [reply datanode/datanode_test.go]: this is automatically generated from "go fmt".

File: subscriber/common/consumer/kafka/kafka.go [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+	offsetReset := sarama.OffsetOldest
+	if jobConfig.StreamingConfig.LatestOffset {
+		offsetReset = sarama.OffsetNewest
+	}
+	cfg.Consumer.Offsets.Initial = offsetReset
+	cfg.Consumer.Return.Errors = true
+	if jobConfig.StreamingConfig.ReblanceTimeoutSec > 0 {
+		cfg.Consumer.Group.Rebalance.Timeout = time.Duration(jobConfig.StreamingConfig.ReblanceTimeoutSec) * time.Second
+	}
+	cfg.Version = getKafkaVersion(jobConfig.StreamingConfig.KafkaVersion)
```
[the remark this answers is not in the document]
Comment (lxning) [reply subscriber/common/consumer/kafka/kafka.go]: this is forced by sarama lib

**Pull request merged**